include/boost/geometry/strategies/geographic/buffer_point_circle.hpp

```diff
@@ -0,0 +1,127 @@
+// Boost.Geometry
+
+// Copyright (c) 2018-2019 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_BUFFER_POINT_CIRCLE_HPP
+#define BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_BUFFER_POINT_CIRCLE_HPP
+
+#include <cstddef>
+
+#include <boost/range.hpp>
+
+#include <boost/geometry/util/math.hpp>
+
+#include <boost/geometry/strategies/buffer.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace buffer
+{
+
+/*!
+\brief Create a circular buffer around a point, on the Earth
+\ingroup strategies
+\details This strategy can be used as PointStrategy for the buffer algorithm.
+    It creates a circular buffer around a point, on the Earth. It can be applied
+    for points and multi_points.
+
+\qbk{
+[heading Example]
+[buffer_point_circle_geographic]
+[buffer_point_circle_geographic_output]
+[heading See also]
+\* [link geometry.reference.algorithms.buffer.buffer_7_with_strategies buffer (with strategies)]
+\* [link geometry.reference.strategies.strategy_buffer_point_circle point_circle]
+\* [link geometry.reference.strategies.strategy_buffer_point_square point_square]
+}
+ */
+template
+<
+    typename FormulaPolicy = strategy::andoyer,
+    typename Spheroid = srs::spheroid<double>,
+    typename CalculationType = void
+>
+class point_circle_geographic
```

Comment:
The naming is slightly inconsistent with already existing geographic strategies. We have:
- `strategy::distance::geographic`
- `strategy::side::geographic`
- `strategy::intersection::geographic_segments`
- `strategy::distance::geographic_cross_track`

So my guess is that this should be called `geographic_point_circle`.

Side note: I noticed that the naming of strategies is inconsistent with the directories structure. Typically we want to have namespaces aligned with directories. Should there be CS-specific namespaces in `strategy` namespace resembling CS-specific directories in *strategies* directory? For example: 
- `strategy::geographic::distance::point_point`
- `strategy::geographic::distance::point_segment`
- `strategy::geographic::intersection::segments`
- `strategy::geographic::side::point_segment`
- `strategy::cartesian::distance::point_point` <=> `strategy::cartesian::distance::pythagoras`
Just a side question for the future.

Reply:
I agree with the name `geographic_point_circle` and also with the side note. Just to add that we probably need to change also names like pythagoras and andoyer to something to remind that it is Euclidean distance and first order approximation of geographic/ellipsoidal distance.

Reply:
We have:

- strategy::buffer::point_circle
- strategy::buffer::point_square
And therefore I selected the fitting:
- strategy::buffer::point_circle_geographic

But yes, it can also be
- strategy::buffer::geographic_point_circle

If you prefer that (and apparently you do) then I will change it.

We can actually not rename existing strategies (pythagoras) because they might be used by users.

Reply:
Right, but we can add another strategy with a different name, deprecate the old one by putting the `#warning` in the strategy header and remove the include from main header `geometry.hpp`.

The above is also true for `strategy::buffer::point_circle` to be consistent it could be renamed to `strategy::buffer::cartesian_point_circle`. But I'd like to propose something different.

Correct me if I'm wrong. In the future other geographic and spherical strategies could be implemented. Furthermore as it is now the interface of `buffer()` would allow passing various-CS strategies at the same time, e.g. one for cartesian, one geographic, yet another spherical, etc. If I'm correct and it's the case I propose to redesign the interface by:
- allowing to pass 1 CS-specific buffer strategy as the last, additional parameter of `buffer()`
- treating current strategies as policies of this strategy
This way the parameters would be consistent by design.

Reply:
Yes, indeed, you are right. It currently would allow that (though we could protect that).
So I can do that later - for now I propose to rename it as you both agreed (didn't rename it yet - will do tonight). That is the last point of this review and then I think it can be merged.